custom_components/keymaster/keymaster.yaml

```diff
@@ -207,15 +207,28 @@ binary_sensor:
         {## Get current day of week, date (integer yyyymmdd), and time (integer hhmm) ##}
         {% set current_day = now.strftime('%a')[0:3] | lower %}
         {% set current_date = now.strftime('%Y%m%d') | int %}
+        {% set current_datestamp = as_timestamp(current_date) | int %}
         {% set current_time = now.strftime('%H%M') | int %}
+        {% set current_timestamp = as_timestamp(now) | int %}
 
         {## Get whether date range toggle is enabled as well as start and end date (integer yyyymmdd) ##}
         {## Determine whether current date is within date range using integer (yyyymmdd) comparison ##}
         {% set is_date_range_enabled = is_state('input_boolean.daterange_LOCKNAME_TEMPLATENUM', 'on') %}
-        {% set start_date = states('input_datetime.start_date_LOCKNAME_TEMPLATENUM').replace('-', '') | int %}
-        {% set end_date = states('input_datetime.end_date_LOCKNAME_TEMPLATENUM').replace('-', '') | int %}
-        {% set is_in_date_range = (current_date >= start_date and current_date <= end_date) %}
-    
+        {% set start_date = state_attr('input_datetime.start_date_LOCKNAME_TEMPLATENUM', 'timestamp') | int %}
+        {% set start_time = (states('input_datetime.start_date_LOCKNAME_TEMPLATENUM')[11:]).replace(':', '') | int %}
+        {% set end_date = state_attr('input_datetime.end_date_LOCKNAME_TEMPLATENUM', 'timestamp') | int %}
+        {% set end_time = (states('input_datetime.end_date_LOCKNAME_TEMPLATENUM')[11:]).replace(':', '') | int %}
+
+        {## If the start and end dates match, including time, then we do a basic check against just today ##}
+        {% if (start_date == end_date) %}
+        {%   set is_in_date_range = (current_datestamp == start_date) %}
+        {## Make backwards compatible such that if start & end date times are midnight, we do a day based range only ##}
+        {% elif (start_date < end_date and start_time == end_time and start_time == 0) %}
```

Comment:
My testing of this logic has shown me that it works similar to the original code

Reply:
See above